esiclient/tests/unit/v1/test_node_network.py

```diff
@@ -26,229 +26,14 @@ def setUp(self):
-            "internal_port": 23,
-            "external_port": 23,
-            "internal_port_id": "neutron_port_uuid_1"
-        })
-
-        def mock_node_get(node_uuid):
-            if node_uuid == "11111111-2222-3333-4444-aaaaaaaaaaaa":
-                return self.node1
-            elif node_uuid == "11111111-2222-3333-4444-bbbbbbbbbbbb":
-                return self.node2
-            return None
-        self.app.client_manager.baremetal.node.get.side_effect = mock_node_get
-
-        def mock_neutron_port_get(port_uuid):
-            if port_uuid == "neutron_port_uuid_1":
-                return self.neutron_port1
-            elif port_uuid == "neutron_port_uuid_2":
-                return self.neutron_port2
-            return None
-        self.app.client_manager.network.get_port.\
-            side_effect = mock_neutron_port_get
-
-        def mock_neutron_port_forwardings(fip):
-            if fip.id == "floating_ip_uuid_1":
-                return [self.pfw1, self.pfw2]
-            return []
-        self.app.client_manager.network.port_forwardings.\
-            side_effect = mock_neutron_port_forwardings
-
-        self.app.client_manager.network.find_network.\
-            return_value = self.network
-        self.app.client_manager.network.get_network.\
-            return_value = self.network
-        self.app.client_manager.network.ports.\
-            return_value = [self.neutron_port1, self.neutron_port2]
-        self.app.client_manager.network.networks.\
-            return_value = [self.network, self.floating_network]
-        self.app.client_manager.baremetal.node.list.\
-            return_value = [self.node1, self.node2]
-        self.app.client_manager.network.ips.\
-            return_value = [self.floating_ip, self.floating_ip_pfw]
-
-    def test_take_action(self):
-        self.app.client_manager.baremetal.port.list.\
-            return_value = [self.port1, self.port2, self.port3]
-
-        arglist = []
-        verifylist = []
-
-        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
-
-        results = self.cmd.take_action(parsed_args)
-        expected = (
-          ["Node", "MAC Address", "Port", "Network", "Fixed IP",
-           "Floating Network", "Floating IP"],
-          [['node1', 'aa:aa:aa:aa:aa:aa', 'neutron_port_1', 'test_network',
-           '1.1.1.1', 'floating_network', '9.9.9.9 (22:22,23:23)'],
-           ['node2', 'bb:bb:bb:bb:bb:bb', None, None, None, None, None],
-           ['node2', 'cc:cc:cc:cc:cc:cc', 'neutron_port_2', 'test_network',
-            '2.2.2.2', 'floating_network', '8.8.8.8']]
-        )
-        self.assertEqual(expected, results)
-        self.app.client_manager.baremetal.port.list.\
-            assert_called_once_with(detail=True)
-
-    def test_take_action_neutron_port_does_not_exist(self):
-        self.app.client_manager.baremetal.port.list.\
-            return_value = [self.port4]
-
+    def test_get_parser(self):
```

Comment:
There's no real need to test this; however, we do need to test `take_action`. Since it's a unit test, what happens is you mock any outside function calls that you make (previously the various openstack `self.app.client_manager.XXXX.YYYYY`) and hard code the results. Then you verify the results, and that the appropriate function calls are made. You also have to test each branch in a conditional, which is why there were so many tests here; however since the updated code shifts all that into into the `esisdk`, we can eliminate those test cases.

That means in this case, we just have to mock `node.network_list`. As a matter of thoroughness though, I'd mock the return value to cover a variety of cases - trunk ports, floating IPs, floating IPs with port forwarding, as well as baremetal ports with nothing associated with them.

If you haven't done so yet, I think you should be able to create trunk ports and floating IPs (the ESI documentation should explain how). I'd recommend doing so and running `esi network list` with the current packaged version of `python-esiclient` to see what it looks like.